src/sentry/static/sentry/app/views/performance/realUserMonitoring/index.tsx

```diff
@@ -0,0 +1,177 @@
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import SentryDocumentTitle from 'app/components/sentryDocumentTitle';
+import {t} from 'app/locale';
+import {PageContent} from 'app/styles/organization';
+import {Organization, Project, GlobalSelection} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import {isAggregateField} from 'app/utils/discover/fields';
+import {decodeScalar} from 'app/utils/queryString';
+import {tokenizeSearch, stringifyQueryObject} from 'app/utils/tokenizeSearch';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import withOrganization from 'app/utils/withOrganization';
+import withProjects from 'app/utils/withProjects';
+
+import RumContent from './content';
+import {getTransactionName} from '../utils';
+import {transactionSummaryRouteWithQuery} from '../transactionSummary/utils';
+
+type Props = {
+  location: Location;
+  organization: Organization;
+  projects: Project[];
+  selection: GlobalSelection;
+} & Pick<WithRouterProps, 'router'>;
+
+type State = {
+  eventView: EventView | undefined;
+};
+
+class RealUserMonitoring extends React.Component<Props> {
+  state: State = {
+    eventView: generateRumEventView(
+      this.props.location,
+      getTransactionName(this.props.location)
+    ),
+  };
+
+  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
+    return {
+      ...prevState,
+      eventView: generateRumEventView(
+        nextProps.location,
+        getTransactionName(nextProps.location)
+      ),
+    };
+  }
+
+  getDocumentTitle(): string {
+    const name = getTransactionName(this.props.location);
+
+    const hasTransactionName = typeof name === 'string' && String(name).trim().length > 0;
+
+    if (hasTransactionName) {
+      return [String(name).trim(), t('RUM')].join(' - ');
+    }
+
+    return [t('Summary'), t('RUM')].join(' - ');
+  }
+
+  renderNoAccess = () => {
+    const {router, organization, location} = this.props;
+
+    if (organization.features.includes('performance-view')) {
+      const transactionName = getTransactionName(this.props.location);
+      if (!transactionName) {
+        // If there is no transaction name, redirect to the Performance landing page
+
+        browserHistory.replace({
+          pathname: `/organizations/${organization.slug}/performance/`,
+          query: {
+            ...location.query,
+          },
+        });
+        return null;
+      }
+
+      return (
+        <Redirect
+          router={router}
+          to={transactionSummaryRouteWithQuery({
+            orgSlug: organization.slug,
+            transaction: transactionName,
+            projectID: decodeScalar(location.query.project),
+            query: location.query,
+          })}
+        />
+      );
+    } else {
+      return <Alert type="warning">{t("You don't have access to this feature")}</Alert>;
```

Comment:
I would likely hoist this up so you can skip the else.

```js
if (!hasFeature) {
  return <Alert ... />
}
// other code
```

⛳

Reply:
Addressed in 33a7282